### PR TITLE
add problem data to instructor reporting

### PR DESCRIPTION
**Pull request by KatelynGit**
### What are the relevant tickets?
https://github.com/mitodl/hq/issues/11464

### Description (What does it do?)

add problem data to new instructor reporting table

### How can this be tested?
dbt build --select instructor_module_problems_report

### CODE REVIEW

File: src/ol_dbt/models/reporting/_reporting__models.yml [hunk 1]
```diff
@@ -1370,3 +1370,33 @@ models:
   - name: user_latest_row_desc
     description: integer, row number indicating the latest row for each user based
       on application and course run dates
+
+- name: instructor_module_problems_report
```
Comment (rachellougee) [inline src/ol_dbt/models/reporting/_reporting__models.yml:1373]: Can we add some tests in this new model? At minimum: not_null on user_email, courserun_readable_id, and problem_name. A unique combined key to prevent fan-out.

File: src/ol_dbt/models/reporting/instructor_module_problems_report.sql [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+from enrollment
+inner join user
+    on enrollment.user_fk = user.user_pk
+inner join course_run
+    on enrollment.courserun_fk = course_run.courserun_pk
+inner join problem
+    on
+        course_run.courserun_readable_id = problem.courserun_readable_id
+        and course_run.platform = problem.platform
+left join problem_events
```
Comment (rachellougee) [inline src/ol_dbt/models/reporting/instructor_module_problems_report.sql:61]: This left join produced a lot of empty rows where attempt, success, and grade are all null have no actionable value

**Review by rachellougee — APPROVED** [review]
dbt runs fine, I left two comments, both are worth addressing before merging

File: src/ol_dbt/models/reporting/instructor_module_problems_report.sql (hunk 2 above)
Comment (KatelynGit) [reply src/ol_dbt/models/reporting/instructor_module_problems_report.sql:61]: they want to see when learners don't complete problems- however I agree it's too many records I'm checking with the requestor- waiting for a reply
